searcher.py

```diff
@@ -45,13 +47,69 @@ def search(self, client, output_path_guest, topic_path_host, topic_path_guest, g
+                sys.exit('Query format unknown...')
+            single_query = queries[:query_end+6]
+            single_query_file = os.path.splitext(os.path.basename(self.config.topic))[0] + '.single.txt'
+            out_file = open(os.path.join(topic_path_host, single_query_file), 'w')
+            out_file.write(single_query)
+            out_file.close()
+
+        # Time empty search
+        search_args['topic']['path'] = os.path.join(topic_path_guest, single_query_file)
+        container = client.containers.run("{}:{}".format(self.config.repo, save_tag),
+                                          command="bash -c 'time /search --json {}'".format(json.dumps(json.dumps(search_args))), volumes=volumes, detach=True)
+        load_times = []
+        for line in container.logs(stream=True):
+            match = re.match('^(real|user|sys)\t*(.*)m(.*)s$', line.decode('utf-8'))
+            if match:
+                load_times.append(match)
+
+        # Time actual search
+        search_args['topic']['path'] = os.path.join(topic_path_guest, os.path.basename(self.config.topic))
         print("Starting container from saved image...")
         container = client.containers.run("{}:{}".format(self.config.repo, save_tag),
-                                          command="sh -c '/search --json {}'".format(json.dumps(json.dumps(search_args))), volumes=volumes, detach=True)
+                                          command="bash -c 'time /search --json {}'".format(json.dumps(json.dumps(search_args))), volumes=volumes, detach=True)
 
+        search_times = []
         print("Logs for search in container with ID {}...".format(container.id))
         for line in container.logs(stream=True):
+            match = re.match('^(real|user|sys)\t*(.*)m(.*)s$', line.decode('utf-8'))
+            if match:
+                search_times.append(match)
             print(str(line.decode('utf-8')), end="")
+        print()
+
+        print('**********')
+        print('Index load timing information')
+        print(load_times[0].group(0))
+        print(load_times[1].group(0))
+        print(load_times[2].group(0))
+        print()
+
+        print('**********')
+        print('Search timing information')
+        print(search_times[0].group(0))
+        print(search_times[1].group(0))
+        print(search_times[2].group(0))
+        print()
+
+        result_minutes = []
+        result_seconds = []
+        for i in range(len(load_times)):
+            result_minutes.append(int(search_times[i].group(2)) - int(load_times[i].group(2)))
+            result_seconds.append(float(search_times[i].group(3)) - float(load_times[i].group(3)))
+        print('**********')
+        print('Search timing less load')
+        print('real\t{}m{:.3f}s'.format(result_minutes[0], result_seconds[0]))
+        print('user\t{}m{:.3f}s'.format(result_minutes[1], result_seconds[1]))
+        print('sys\t{}m{:.3f}s'.format(result_minutes[2], result_seconds[2]))
```

Comment:
Small nit can a space be added so the formatting is consistent between the output of `time`? Might be useful for parsing the output later if needed.

```
**********
Search timing information
real    0m 13.72s
user    0m 17.58s
sys     0m 0.43s

**********
Search timing less load
real    0m12.680s
user    0m15.300s
sys     0m0.330s
```